src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -1056,11 +1055,11 @@ public bool UnloadDirectly(StringSegment databaseName, IdleDatabaseActivity idle
 
                 // DateTime should be only null in tests
                 if (idleDatabaseActivity is { DateTime: not null })
-                    _wakeupTimers.TryAdd(databaseName.Value, new Timer(
+                    _wakeupTimers.GetOrAdd(databaseName.Value, _ => new Timer(
                         callback: _ => NextScheduledActivityCallback(databaseName.Value, idleDatabaseActivity),
                         state: null,
                         // in case the DueTime is negative or zero, the callback will be called immediately and database will be loaded.
-                        dueTime: idleDatabaseActivity.DueTime > 0 ? idleDatabaseActivity.DueTime : 0,
+                        dueTime: idleDatabaseActivity.DueTime,
```

Comment:
if `idleDatabaseActivity.DueTime` is negative, this will throw no ?

Reply:
cannot be negative: https://github.com/ravendb/ravendb/pull/19879